Description: skill p option
 Make the p option work again
Author: Craig Small <csmall@dropbear.xyz>
Origin: upstream
Bug-Debian: https://bugs.debian.org/1025915
Applied-Upstream: 4.0.3
Reviewed-by: Craig Small <csmall@debian.org>
Last-Update: 2022-12-19
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/skill.c
+++ b/src/skill.c
@@ -287,6 +287,8 @@
     for (i=0; i < total_procs; i++) {
         if (PIDS_GETINT(PID) == my_pid || PIDS_GETINT(PID) == 0)
             continue;
+        if (pids && !match_intlist(PIDS_GETINT(PID), pid_count, pids))
+            continue;
         if (uids && !match_intlist(PIDS_GETUNT(EUID), uid_count, (int *)uids))
             continue;
         if (ttys && !match_intlist(PIDS_GETINT(TTY), tty_count, ttys))
